adrs/0002-monorepo.md

```diff
@@ -0,0 +1,34 @@
+# [Transition Stacks to a monorepo]
+
+**Date:** 2022-11-28
+
+**Related PR:** #1191
+
+## Issue
+
+The Stacks repo is growing in complexity and will soon require many different dependencies that are optional for most consumers.
+
+## Decision
+
+Despite shipping many different concepts (atomic classes, CSS components, JS components), the interdependency chains are relatively linear. This means that we can fairly trivially split the repo into smaller portions.
+
+Our current codebase could be split like so:
+- Atomic classes (no dependencies)
+- CSS Components (depends on Atomic components)
```

Comment:
I would prefer to keep CSS components and JS components in the same package (in the spirit of a true component based architecture - see also [folder-by-feature vs folder-by-type](https://softwareengineering.stackexchange.com/questions/338597/folder-by-type-or-folder-by-feature)).  I would even go a step further and create folders for individual components where we keep style (less), behavior (stimulus - when applicable) and component test files (tbd - perhaps wtr + testing-library) together.
In an ideal scenario even the "docs" would be in the same component folder but I understand that it won't be an easy thing to achieve given our current setup. I would be ok to settle for docs in a separate package (revaluate perhaps at a later stage).

Reply:
The reason I had suggested we separate the two is two-fold:
1. If you only need the Stacks classes and not the JS components, you don't have to pull them in
2. We're looking to modernize our JS component story, which means potentially having more than one supported component library (e.g. Stimulus + Web Components / React / Svelte / whatever), at least for a short while

That being said, I'm open to discussion on this point. I'm very much interested in hanging with the team, hashing out the pros / cons and coming to a hard decision. I'll schedule something for us for before the end of the calendar year.

Reply:
Point 1 could be achieved even if we ship a single npm "stacks-components" package. 
We will still have a way for consumers to opt in for css only if they want to I suppose.

The reason why I am suggesting for the component styles to go hand in hand with Stimulus is because I don't see us reusing any of that source code (Stimulus and Less) when we will build web components (or anything other component library). 
My thinking as of now is that new components will depend exclusively on the atomic classes package and work in any context without depending on global styles being applied to a page or not.

I believe styles will have to be rewritten but I might be miscalculating things. Discovery will certainly help.
In case we will reuse the same component styles we have today in Stacks with the new components then I would understand why we would keep components styles in a separate package.

Reply:
It's just my ¢2, but I agree with Giamir that a single package for both CSS components and JS components makes the most sense. 

On top of Giamir's points, it's plausible that we could convert a CSS component to a JS component at some point. Correct me if I'm wrong, but I imagine we'd need to deprecate the component in the CSS package and port it to the JS component package. That seems like it would be messy and a single Stacks components package makes more sense to me intuitively.

Reply:
I think this is the only pending topic before we could merge this ADR too. 🙂

Reply:
This is actually something I feel strongly about and would be better suited to discuss over a hangout. In the meanwhile, I'm going to go ahead and merge as-is, since that line is just a suggestion and implementation can differ from what I have here.

We can always come back and update this later. The PR for that will serve as its own little mini ADR, since it'll contain the reasoning for the change.

Reply:
I have mixed feelings about this PR being merged in without reaching an agreement on this point first. This is primarily because what we merge we consider automatically as approved (we don't have a status flag). 
Can I suggest to add a comment/line directly in the ADR content to remind ourselves that further discussion is required about packages organisation.
Don't get me wrong, I like to move things forward fast but I also think it's ok to slow down a bit when we cannot agree on something as a team. @b-kelly Can I leave this small adjustment with you together with scheduling a meeting for us to talk further about this topic when the time is right? Thanks and sorry if this request sounds a bit pedantic. 🙂

Reply:
@giamir I'll make an adjustment. I see this section as a "for example" section, not as something I was proposing in this PR. The wording `could be` alludes to that. Would it make more sense if I reworded it like:

> For example, our current codebase could be split like so

Reply:
Sounds good to me. Thanks. 
I guess the more important action item is to come together as a team, take a decision and retrospectively justify our reasoning in this ADR too (we are not in a rush though). 🙂

Reply:
Wording tweaked in 84a917cfa59ca69c789b4e425406de6a3448227f